### PR TITLE
Define terminal menu items using GLib.MenuModel

### PR DESCRIPTION
**Pull request by colinkiama**
Replaces usage of Gtk.MenuItem with GLib.MenuModel in the Terminal widget as #1157

### CODE REVIEW

File: src/Widgets/Terminal.vala [hunk 1]
```diff
@@ -36,29 +46,44 @@ public class Code.Terminal : Gtk.Box {
...
                 menu.select_first (false);
                 menu.popup_at_pointer (event);
             }
             return false;
         });
 
+        realize.connect (() => {
+            current_clipboard = terminal.get_clipboard (Gdk.SELECTION_CLIPBOARD);
+            copy_action.set_enabled (terminal.get_has_selection ());
+
```
Comment (danirabbit) [inline src/Widgets/Terminal.vala:80]: extra whitespace

File: src/Widgets/Terminal.vala [hunk 2]
```diff
@@ -74,6 +99,14 @@ public class Code.Terminal : Gtk.Box {
...
     }
 
+    public void copy () {
+        terminal.copy_clipboard ();
+    }
+
+    public void paste () {
+        terminal.paste_clipboard ();
+    }
+
```
Comment (danirabbit) [inline src/Widgets/Terminal.vala:109]: Is there a reason to add new public methods here? Seems like this is simple enough we could just keep it as lambdas as before

File: src/Widgets/Terminal.vala [hunk 3]
```diff
@@ -5,13 +5,23 @@
...
     private const string LEGACY_SETTINGS_SCHEMA = "org.pantheon.terminal.settings";
     private const string SETTINGS_SCHEMA = "io.elementary.terminal.settings";
 
-    private GLib.Pid child_pid;
     public Vte.Terminal terminal { get; construct; }
+    public SimpleActionGroup actions { get; construct; }
+
+    private GLib.Pid child_pid;
+    private SimpleAction copy_action;
+    private SimpleAction paste_action;
```
Comment (danirabbit) [inline src/Widgets/Terminal.vala:23]: Do we need to hold a reference for these actions here? It seems like they're only used in the construct method right?

**Review by danirabbit — CHANGES_REQUESTED** [review]
Looks good and works as expected! Just a couple of comments

File: src/Widgets/Terminal.vala (hunk 2 above)
Comment (colinkiama) [reply src/Widgets/Terminal.vala]: No reason. I moved the logic back in to lambdas.

File: src/Widgets/Terminal.vala (hunk 3 above)
Comment (colinkiama) [reply src/Widgets/Terminal.vala]: Thanks for catching that! We don't. These have been moved to the construct block.

**Review by danirabbit — APPROVED** [review]
Nice work :tada:

**Pull request merged**